example/EmbeddedChessPieces/src/ECPColorDetection/ECPColorDetection.h

```diff
@@ -13,32 +13,96 @@
 #define ECPColorDetection_h
 
 #include <Dezibot.h>
-
-#define IS_WHITE_FIELD_THRESHOLD 100.0
+#include <ECPSignalDetection/ECPSignalDetection.h>
 
 #define COLOR_CORRECTION_LIGHT_R 43
 #define COLOR_CORRECTION_LIGHT_G 33
 #define COLOR_CORRECTION_LIGHT_B 35
 
-class ECPColorDetection {
-protected:
-    Dezibot &dezibot;
+/**
+ * @brief Color of field the dezibot is standing on.
+ * 
+ * Convertible to int, i.e. BLACK_FIELD == 0, WHITE_FIELD == 1
+ * 
+ */
+enum FieldColor {
+    BLACK_FIELD, 
+    WHITE_FIELD,
+    UNAMBIGUOUS
+};
 
+class ECPColorDetection {
 public:
-    ECPColorDetection(Dezibot &d);
+    ECPColorDetection(Dezibot &d, ECPSignalDetection &ir);    
 
     /**
-     * @brief Determine if brightness value represents a white or a black chess
+     * @brief Calibrate threshold for white and black field using color sensor.
+     * 
+     */
+    void calibrateFieldColor();
+
+    /**
+     * @brief Calibrate threshold for white and black field using infrared.
+     * 
+     */
+    void calibrateIRFieldColor();
+
+    /**
+     * @brief Determine if measured value represents a white or black chess
      *        field.
+     * 
+     * @details Uses color detection mode of \p useInfraredColorDetection flag.
+     *          Default is color detection using color sensor.
      *
-     * Note that the room must be well-lit or color correction light is turned on
-     * (see \see turnOnColorCorrectionLight).
-     * At a normalized ambient light of about 10.0 or lower white field will be interpreted as black.
+     * Note that the field color detection should be calibrated for selected mode.
+     * @see calibrateFieldColor, calibrateIRFieldColor
      * 
-     * @return true if surface is white-ish
-     * @return false if surface is black-ish
+     * @return Determined field color
      */
-    bool isWhiteField();
+    FieldColor getFieldColor();
+
+    /**
+     * @brief Calculate likely field color.
+     * 
+     * @details Uses color detection mode of \p useInfraredColorDetection flag.
+     *          Default is color detection using color sensor.
+     * 
+     * Usecase: the determined field color is unambiguous but a likely color is need
+     * 
+     * Determine if measured value is closer to the black or white field threshold
+     * In case the distance is identical black field is preferred
+     * 
+     * @return Likely fieldColor, only BLACK_FIELD and WHITE_FIELD possible
```

Comment:
```suggestion
     * @return FieldColor likely field color, only BLACK_FIELD and WHITE_FIELD possible
```
It' always `@return <RETURN_TYPE> <description>` ;)

Reply:
Fixed in [57e7495](https://github.com/nicosrm/24-emb-chess/pull/40/commits/57e74956b0712787797ea2ea4d639005a3940e98)